core/src/main/scala/kafka/server/KafkaApis.scala

```
@@ -2926,16 +2926,52 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
-  def handleListClientMetricsResources(request: RequestChannel.Request): Unit = {
-    val listClientMetricsResourcesRequest = request.body[ListClientMetricsResourcesRequest]
+  /**
+   * Handle ListConfigResourcesRequest. If resourceTypes are not specified, it uses ListConfigResourcesRequest#defaultResourceTypes
+   * to retrieve config resources. If resourceTypes are specified, it returns matched config resources.
+   * An unknown config resource type is ignored.
+   */
+  private def handleListConfigResources(request: RequestChannel.Request): Unit = {
+    val listConfigResourcesRequest = request.body[ListConfigResourcesRequest]
 
     if (!authHelper.authorize(request.context, DESCRIBE_CONFIGS, CLUSTER, CLUSTER_NAME)) {
-      requestHelper.sendMaybeThrottle(request, listClientMetricsResourcesRequest.getErrorResponse(Errors.CLUSTER_AUTHORIZATION_FAILED.exception))
+      requestHelper.sendMaybeThrottle(request, listConfigResourcesRequest.getErrorResponse(Errors.CLUSTER_AUTHORIZATION_FAILED.exception))
     } else {
-      val data = new ListClientMetricsResourcesResponseData().setClientMetricsResources(
-        clientMetricsManager.listClientMetricsResources.stream.map(
-          name => new ClientMetricsResource().setName(name)).collect(Collectors.toList()))
-      requestHelper.sendMaybeThrottle(request, new ListClientMetricsResourcesResponse(data))
+      val data = new ListConfigResourcesResponseData()
+
+      var resourceTypes = listConfigResourcesRequest.data().resourceTypes()
+      if (resourceTypes.isEmpty) {
+        resourceTypes = listConfigResourcesRequest.defaultResourceTypes()
+      }
+
+      val result = new util.ArrayList[ListConfigResourcesResponseData.ConfigResource]()
+      if (resourceTypes.contains(ConfigResource.Type.GROUP.id)) {
```

One final change please. Given that we've agreed that v1 supports exactly the current set of supported config resource types (Topic, Client metrics, Broker logger, Broker and Topic), if the request includes any outside this set, the request should fail with `UNSUPPORTED_VERSION` just like `handleAlterIncrementalConfigs`. Then I think it's nice and tight. Both client and broker are in agreement about what the valid set of resource types is. No surprises.

---

To align with `handleIncrementalAlterConfigsRequest`:
* Change the handler to return `UNSUPPORTED_VERSION` if there is a config resource type not in supported list.
* Change `ListConfigResourcesRequest#defaultResourceTypes` to `ListConfigResourcesRequest#supportedResourceTypes`.
* Change `KafkaApisTest#testListConfigResourcesV1WithUnknown` to get `UNSUPPORTED_VERSION` error code.